packages/core/src/modules/connections/DidExchangeProtocol.ts

```diff
@@ -523,64 +523,73 @@ export class DidExchangeProtocol {
+          }
 
-      // JWS payload must be base64url encoded
-      const base64UrlPayload = base64ToBase64URL(didRotateAttachment.data.base64)
-      const signedDid = TypedArrayEncoder.fromBase64(base64UrlPayload).toString()
+          // JWS payload must be base64url encoded
+          const base64UrlPayload = base64ToBase64URL(didRotateAttachment.data.base64)
+          const signedDid = TypedArrayEncoder.fromBase64(base64UrlPayload).toString()
 
-      if (signedDid !== message.did) {
-        throw new CredoError(
-          `DID Rotate attachment's did ${message.did} does not correspond to message did ${message.did}`
-        )
-      }
-
-      const { isValid, signerKeys } = await this.jwsService.verifyJws(agentContext, {
-        jws: {
-          ...jws,
-          payload: base64UrlPayload,
-        },
-        jwkResolver: ({ jws: { header } }) => {
-          if (typeof header.kid !== 'string' || !isDid(header.kid, 'key')) {
-            throw new CredoError('JWS header kid must be a did:key DID.')
+          if (signedDid !== message.did) {
+            throw new CredoError(
+              `DID Rotate attachment's did ${message.did} does not correspond to message did ${message.did}`
+            )
           }
 
-          const didKey = DidKey.fromDid(header.kid)
-          return getJwkFromKey(didKey.key)
-        },
-      })
-
-      if (!isValid || !signerKeys.every((key) => invitationKeysBase58?.includes(key.publicKeyBase58))) {
-        throw new DidExchangeProblemReportError(
-          `DID Rotate signature is invalid. isValid: ${isValid} signerKeys: ${JSON.stringify(
-            signerKeys
-          )} invitationKeys:${JSON.stringify(invitationKeysBase58)}`,
-          {
-            problemCode: DidExchangeProblemReportReason.ResponseNotAccepted,
+          const { isValid, signerKeys } = await this.jwsService.verifyJws(agentContext, {
+            jws: {
+              ...jws,
+              payload: base64UrlPayload,
+            },
+            jwkResolver: ({ jws: { header } }) => {
+              if (typeof header.kid !== 'string' || !isDid(header.kid, 'key')) {
+                throw new CredoError('JWS header kid must be a did:key DID.')
+              }
+
+              const didKey = DidKey.fromDid(header.kid)
+              return getJwkFromKey(didKey.key)
+            },
+          })
+
+          if (!isValid || !signerKeys.every((key) => invitationKeysBase58?.includes(key.publicKeyBase58))) {
+            throw new DidExchangeProblemReportError(
+              `DID Rotate signature is invalid. isValid: ${isValid} signerKeys: ${JSON.stringify(
+                signerKeys
+              )} invitationKeys:${JSON.stringify(invitationKeysBase58)}`,
+              {
+                problemCode: DidExchangeProblemReportReason.ResponseNotAccepted,
+              }
+            )
           }
-        )
+        } catch (e) {
+          this.logger.warn(`Document does not contain didRotate. Error: ${e.message}`)
```

Comment:
I think this warning should be shown when there is not any did rotate attachment (i.e. in an `else` block in this case), but I don't see why we should be catching the validations we do here (like the did, JWS header, etc.)